File: 45/py/45.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        if len(nums) <= 1 :
            return 0

        minStep = 0
        curFarthest = 0
        curBound = 0

        for i in range(0, len(nums)-1, 1) :
            if curFarthest < i + nums[i] :
                curFarthest = i + nums[i]

            if i == curBound :
                curBound = curFarthest
                minStep += 1

                if curBound >= len(nums) - 1 :
                    return minStep

        return minStep

File: 45/go/45.go
func jump(nums []int) int {

    // check length
    if len(nums) <= 1 {
        return 0;
    }


    // Greedy
    // check every step

    minSteps := 0
    curFarthest := 0
    curBound := 0

    for i, n := range nums {
        if curFarthest < i + n {
            curFarthest = i + n
        }

        if i == curBound {
            curBound = curFarthest
            minSteps += 1
            if curBound >= len(nums) - 1{
                return minSteps
            }
        }
    }

    return minSteps
}